yeonjae/programmers-1/제일작은수제거하기.js

```diff
@@ -0,0 +1,17 @@
+function solution(arr) {
+  const minVal = Math.min(...arr);
+
+  for (let i = 0; i < arr.length; i++) {
+    if (arr.length === 1) return [-1];
+    else if (arr[i] === minVal) {
+      arr.splice(i, 1);
+    }
+  }
+  return arr;
+}
```

Comment:
```suggestion
function solution(arr) {
  const minVal = Math.min(...arr);

  for (let i = 0; i < arr.length; i++) {
    if (arr.length === 1) return [-1];
    if (arr[i] === minVal) arr.splice(i, 1);
  }
  return arr;
}
```
- 잘푸셨습니다! 고생하셨어요 👍어차피 상단의 if문에서 조건식이 되면 return되기 때문에 else if보다는 그냥 if문으로 가져가셔도 좋을 것 같습니다 :) 
- 아래 풀이는 for문이 아닌 findIndex로 찾고자하는 요소의 인덱스를 찾는 방법으로 결과를 도출하였습니다 참고해보세요!
```js
function solution(arr) {
  const minNum = Math.min(...arr);
  const minIdx = arr.findIndex((num) => num === minNum);

  if (arr.length === 1) return [-1];
  arr.splice(minIdx, 1);

  return arr;
}
```
- 아래 코드는 filter를 이용해서 findIndex와 splice를 써서 비교적 안좋은 효율을 가진 위의 코드를 개선한 코드입니다.
- findIndex와 splice는 최악의 경우 둘다 O(N)의 시간복잡도를 갖기때문에 최종적으로 O(2N)을 갖지만, 아래 코드는 단 1번의 filter로 결과물을 도출할 수 있기 때문에 더 효율적이라고 할 수 있겠습니다 🙆‍♂️
```js
function solution(arr) {
  const minNum = Math.min(...arr);
  const result = arr.filter((el) => el !== minNum);

  return result.length === 0 ? [-1] : result;
}
```

Reply:
오 두가지모두 확인했씁니다 ! 추가적인 풀이 감사해요🤠🤠